setup.py

```diff
@@ -294,6 +294,11 @@ def get_long_description():
             ]
         },
         install_requires=["apptools", "setuptools", "traits"],
+        extras_require={
+            "ipykernel": ["ipykernel"],
+            "pyface": ["pyface"],
+            "traitsui": ["traitsui"]
```

Comment:
If I understand correctly, usually `extras_require` is specified in terms of a context instead of the package e.g. `ipykernel` is a dependency for the "ipython" integration in envisage, `pyface` and `traitsui` are dependencies of the `tasks` or `ui` component of envisage.

```suggestion
            "ipython": ["ipykernel"],
            "ui": ["pyface", "traitsui"],
```